rectools/models/sasrec.py

```diff
@@ -1184,28 +1361,28 @@ def __init__(  # pylint: disable=too-many-arguments, too-many-locals
         self.lr = lr
         self.loss = loss
         self.gbce_t = gbce_t
+        self.top_k_saved_val_reco = top_k_saved_val_reco
 
-    def _fit(
-        self,
-        dataset: Dataset,
-    ) -> None:
-        processed_dataset = self.data_preparator.process_dataset_train(dataset)
-        train_dataloader = self.data_preparator.get_dataloader_train(processed_dataset)
+    def _fit(self, dataset: Dataset) -> None:
+        self.data_preparator.process_dataset_train(dataset)
+        train_dataloader = self.data_preparator.get_dataloader_train()
+        val_dataloader = self.data_preparator.get_dataloader_val()
 
         torch_model = deepcopy(self._torch_model)  # TODO: check that it works
-        torch_model.construct_item_net(processed_dataset)
+        torch_model.construct_item_net(self.data_preparator.train_dataset)
 
-        n_item_extra_tokens = self.data_preparator.n_item_extra_tokens
         self.lightning_model = self.lightning_module_type(
             torch_model=torch_model,
             lr=self.lr,
             loss=self.loss,
             gbce_t=self.gbce_t,
-            n_item_extra_tokens=n_item_extra_tokens,
+            data_preparator=self.data_preparator,
+            verbose=self.verbose,
+            top_k_saved_val_reco=self.top_k_saved_val_reco,
         )
 
         self.fit_trainer = deepcopy(self._trainer)
-        self.fit_trainer.fit(self.lightning_model, train_dataloader)
+        self.fit_trainer.fit(self.lightning_model, train_dataloader, val_dataloader)
```

Comment:
We should clean data_preparator processed data after this

Reply:
we agreed that it would be in the next PR